File: server/src/weblab/data/dto/ExperimentUse.py
#!/usr/bin/env python
#-*-*- encoding: utf-8 -*-*-
#
# Copyright (C) 2005-2009 University of Deusto
# All rights reserved.
#
# This software is licensed as described in the file COPYING, which
# you should have received as part of this distribution.
#
# This software consists of contributions made by many individuals, 
# listed below:
#
# Author: Jaime Irurzun <jaime.irurzun@gmail.com>
# 

class ExperimentUse(object):
    
    def __init__(self, start_date, end_date, experiment, agent, origin, id=None):
        self.id         = id
        self.start_date = start_date
        self.end_date   = end_date
        self.experiment = experiment
        self.agent      = agent # User or ExternalEntity
        self.origin     = origin

    def __repr__(self):
        return u"ExperimentUse(id = %s, start_date = '%s', end_date = '%s', experiment = %r, agent = %r, origin = '%s')" % (
                self.id,
                self.start_date,
                self.end_date,
                self.experiment,
                self.agent,
                self.origin
            ) 

File: gwt-client/src/es/deusto/weblab/client/dto/experiments/ExperimentUse.java
/*
* Copyright (C) 2005 onwards University of Deusto
* All rights reserved.
*
* This software is licensed as described in the file COPYING, which
* you should have received as part of this distribution.
*
* This software consists of contributions made by many individuals, 
* listed below:
*
* Author: Jaime Irurzun <jaime.irurzun@gmail.com>
*
*/

package es.deusto.weblab.client.dto.experiments;

import java.util.Date;

import es.deusto.weblab.client.dto.users.Agent;

public class ExperimentUse {
	
	private int id;
	private Date startDate;
	private Date endDate;
	private Experiment experiment;
	private Agent agent;
	private String origin;
	
	public ExperimentUse() {}
	
	public ExperimentUse(Date startDate, Date endDate, Agent agent, Experiment experiment, String origin) {
		this.agent = agent;
		this.experiment = experiment;
		this.startDate = startDate;
		this.endDate = endDate;
		this.origin = origin;
	}
		
	public int getId() {
		return this.id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public Date getStartDate() {
		return this.startDate;
	}
	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}
	public Date getEndDate() {
		return this.endDate;
	}
	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}
	public void setExperiment(Experiment experiment) {
		this.experiment = experiment;
	}
	public Experiment getExperiment() {
		return this.experiment;
	}
	public String getOrigin() {
		return this.origin;
	}
	public void setOrigin(String origin) {
		this.origin = origin;
	}
	public void setAgent(Agent agent) {
		this.agent = agent;
	}
	public Agent getAgent() {
		return this.agent;
	}
}
